modules/fx/arrow-fx/src/main/kotlin/arrow/fx/typeclasses/ParApplicative.kt

```diff
@@ -20,6 +20,8 @@ internal fun <F> Concurrent<F>.ParApplicative(ctx: CoroutineContext? = null): Ap
   override fun <A, B> Kind<F, A>.ap(ff: Kind<F, (A) -> B>): Kind<F, B> =
     _ctx.parMapN(ff, this@ap) { f, a -> f(a) }
 
+  override fun <A, B> Kind<F, A>.lazyAp(ff: () -> Kind<F, (A) -> B>): Kind<F, B> = ap(ff())
```

Comment:
@nomisRev This is needed to have `parTraverse` pass tests. This implies the old behaviour of `traverse` where the traversable functor gets folded first to build up an `IO` action which then gets executed. Is this a necessity when working with `parTraverse` or can we somehow change this to a step by step process as well? This would mean having a sensible implementation of `lazyAp` which still preserves `parTraverse`'s functionality. (The benefit would be the same as the benefits `traverse` gets from using `lazyAp`)